server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -542,6 +567,7 @@ protected Query getPrefixQuery(String field, String termStr) throws ParseExcepti
         }
         List<Query> queries = new ArrayList<>();
         for (Map.Entry<String, Float> entry : fields.entrySet()) {
+            onDiscoveredField(entry.getKey());
```

Comment:
Not needed, `getPrefixQuerySingle` discovers fields

Reply:
done